crates/rover-client/src/error.rs

```diff
@@ -0,0 +1,30 @@
+use thiserror::Error;
+
+/// An enum represent all type of possible failures.
+#[derive(Error, Debug)]
+pub enum RoverClientError {
+    /// The provided GraphQL was invalid.
+    #[error("encountered a GraphQL error, registry responded with: {msg}")]
+    GraphQLError {
+        /// The error message.
+        msg: String,
+    },
+    /// Tried to build a [HeaderMap] with an invalid header value.
+    #[error("header value was not valid: {msg:?}")]
+    HeadersError {
+        /// The error message.
+        msg: http::header::InvalidHeaderValue,
+    },
```

Comment:
```suggestion
    #[error("invalid header value")]
    InvalidHeader {#[from] http::header::InvalidHeaderValue},
```


this will automatically coerce any `InvalidHeaderValue` into our custom error type and you don't have to worry about passing the message manually